extensions/panache/hibernate-orm-panache/runtime/src/main/java/io/quarkus/hibernate/orm/panache/runtime/JpaOperations.java

```diff
@@ -268,6 +269,13 @@ public static Object findById(Class<?> entityClass, Object id, LockModeType lock
         return new PanacheQueryImpl(em, jpaQuery, findQuery, params);
     }
 
+    @SuppressWarnings("rawtypes")
+    public static PanacheQuery<?> findAll(Class<?> entityClass, CriteriaQuery<?> criteriaQuery) {
+        String query = "FROM " + getEntityName(entityClass);
+        EntityManager em = getEntityManager();
+        return new PanacheQueryImpl(em, em.createQuery(criteriaQuery), query, null);
```

Comment:
This looks just wrong. I don't think `PanacheQuery.count()` will work without using the correct count query. A criteria query can filter and so it is not an "all" query.

Reply:
Count ex:
`CriteriaBuilder builder = em.getCriteriaBuilder();
CriteriaQuery<Long> countCriteria = builder.createQuery(Long.class);
Root<Entity> entityRoot = countCriteria.from(criteriaQuery.getResultType());
countCriteria.select(builder.count(entityRoot));
countCriteria.where(criteriaQuery.getRestriction());
return em.createQuery(countCriteria).getSingleResult();`